dbt/adapters/glue/impl.py

```diff
@@ -593,29 +593,61 @@ def hudi_write(self, write_mode, session, target_relation, custom_location):
+            'hoodie.datasource.write.recordkey.field': primary_key,
+            'hoodie.table.name': target_relation.name,
+            'hoodie.datasource.hive_sync.database': target_relation.schema,
+            'hoodie.datasource.hive_sync.table': target_relation.name,
+            'hoodie.datasource.hive_sync.enable': 'true',
+        }
+
         if partition_key:
             partition_list = ','.join(partition_key)
-            hudi_partitionning = f''' 'hoodie.datasource.write.partitionpath.field': '{partition_list}', 'hoodie.datasource.hive_sync.partition_extractor_class': 'org.apache.hudi.hive.MultiPartKeysValueExtractor', 'hoodie.datasource.hive_sync.partition_fields': '{partition_list}','hoodie.index.type': 'GLOBAL_BLOOM', 'hoodie.bloom.index.update.partition.path': 'true','''
+            partition_config = {
+                'hoodie.datasource.write.partitionpath.field': f'{partition_list}',
+                'hoodie.datasource.hive_sync.partition_extractor_class': 'org.apache.hudi.hive.MultiPartKeysValueExtractor',
+                'hoodie.datasource.hive_sync.partition_fields': f'{partition_list}',
+                'hoodie.index.type': 'GLOBAL_BLOOM',
+                'hoodie.bloom.index.update.partition.path': 'true',
+            }
         else:
-            hudi_partitionning = ''
-
-        begin_of_hudi_setup = f'''combinedConf = {{'className' : 'org.apache.hudi', 'hoodie.datasource.hive_sync.use_jdbc':'false', 'hoodie.datasource.write.precombine.field': 'update_hudi_ts', 'hoodie.consistency.check.enabled': 'true', 'hoodie.datasource.write.recordkey.field': '{primary_key}', 'hoodie.table.name': '{target_relation.name}', 'hoodie.datasource.hive_sync.database': '{target_relation.schema}', 'hoodie.datasource.hive_sync.table': '{target_relation.name}', 'hoodie.datasource.hive_sync.enable': 'true','''
-
-        hudi_no_partition = f''' 'hoodie.datasource.hive_sync.partition_extractor_class': 'org.apache.hudi.hive.NonPartitionedExtractor', 'hoodie.datasource.write.keygenerator.class': 'org.apache.hudi.keygen.NonpartitionedKeyGenerator','hoodie.index.type': 'GLOBAL_BLOOM', 'hoodie.bloom.index.update.partition.path': 'true','''
+            partition_config = {
+                'hoodie.datasource.hive_sync.partition_extractor_class': 'org.apache.hudi.hive.NonPartitionedExtractor',
+                'hoodie.datasource.write.keygenerator.class': 'org.apache.hudi.keygen.NonpartitionedKeyGenerator',
+                'hoodie.index.type': 'GLOBAL_BLOOM',
+                'hoodie.bloom.index.update.partition.path': 'true',
+            }
 
-        hudi_upsert = f''' 'hoodie.upsert.shuffle.parallelism': 20, 'hoodie.datasource.write.operation': 'upsert', 'hoodie.cleaner.policy': 'KEEP_LATEST_COMMITS', 'hoodie.cleaner.commits.retained': 10}}'''
-
-        hudi_insert = f''' 'hoodie.bulkinsert.shuffle.parallelism': 20, 'hoodie.datasource.write.operation': 'bulk_insert'}}'''
+        if isTableExists:
+            write_mode = 'Overwrite'
```

Comment:
@jduggin-Parchment why was this changed from 'Append' to 'Overwrite' write mode if the table exists? 

It is causing issues as mentioned in https://github.com/aws-samples/dbt-glue/issues/144.